quirk_echo_launcher/src/launcher/virtualized.rs

```diff
@@ -0,0 +1,202 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::launcher::GuestInstance;
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use clap::Parser;
+use command_fds::tokio::CommandFdAsyncExt;
+use log::info;
+use oak_channel::Write;
+use std::{
+    fs,
+    net::Shutdown,
+    os::unix::{io::AsRawFd, net::UnixStream},
+    path::PathBuf,
+    process::Stdio,
+};
+
+const PAGE_SIZE: usize = 4096;
+
+/// Represents parameters used for launching VM instances.
+#[derive(Parser, Clone, Debug, PartialEq)]
+pub struct Params {
+    /// Path to the VMM binary to execute.
+    #[arg(long, value_parser = Params::path_exists)]
+    pub vmm_binary: PathBuf,
+
+    /// Path to the enclave binary to load into the VM.
+    #[arg(long, value_parser = Params::path_exists)]
+    pub enclave_binary: PathBuf,
+
+    /// Path to the Oak Functions application binary to be loaded into the enclave.
+    #[arg(long, value_parser = Params::path_exists)]
+    pub app_binary: PathBuf,
```

Comment:
AFAICT this is really the only value that would differ from the current implementation, right?

Reply:
Factored out shared package to be used in both launchers.